app/src/main/java/cash/bchd/android_neutrino/SendActivity.java

```diff
@@ -180,103 +162,98 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
             }
         });
 
-        Button send = (Button) findViewById(R.id.sendBtn);
-        send.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                TextInputLayout layout = (TextInputLayout) findViewById(R.id.addressLayout);
-                try {
-                    boolean inputError = false;
-                    boolean valid = wallet.validateAddress(address.getText().toString());
-                    if (!valid) {
-                        layout.setError("INVALID BITCOIN CASH ADDRESS");
-                        inputError = true;
-                    } else {
-                        layout.setError(null);
-                    }
-                    Amount toSpend;
-                    if (inputAmount.getText().toString().equals("")) {
-                        TextInputEditText etAmount = (TextInputEditText) findViewById(R.id.amountInput);
-                        etAmount.setError("INVALID AMOUNT");
-                        inputError = true;
-                    }
-                    if (inputError) {
-                        return;
-                    }
-                    if (showingFiat) {
-                        double fiatAmount = Double.valueOf(inputAmount.getText().toString().replace(",", "."));
-                        toSpend = new Amount(ExchangeRates.getInstance().convertToBCH(fiatAmount, Currency.getInstance(fiatCurrency)));
-                    } else {
-                        double bchAmount = Double.valueOf(inputAmount.getText().toString().replace(",", "."));
-                        toSpend = new Amount(bchAmount);
-                    }
-                    if (toSpend.getSatoshis() == 0 || toSpend.getSatoshis() > balance.getSatoshis()) {
-                        TextInputEditText etAmount = (TextInputEditText) findViewById(R.id.amountInput);
-                        etAmount.setError("INVALID AMOUNT");
-                        return;
-                    }
-                    byte[] serializedTx = null;
-                    long txFee = 0;
-                    List<Long> inputVals = null;
-                    if (sendAll) {
-                        Api.SweepAccountResponse tx = wallet.sweepAccount(address.getText().toString(), satPerByte);
-                        serializedTx = tx.getSerializedTransaction().toByteArray();
-                        txFee = tx.getFee();
-                        inputVals = tx.getInputValuesList();
+        Button send = findViewById(R.id.sendBtn);
+        send.setOnClickListener(v -> {
+            TextInputLayout layout = findViewById(R.id.addressLayout);
+            try {
+                boolean inputError = false;
+                boolean valid = wallet.validateAddress(address.getText().toString());
+                if (!valid) {
+                    layout.setError("INVALID BITCOIN CASH ADDRESS");
```

Comment:
Shouldn't this string be in the XML below?

Reply:
yes, to all your comments with strings, I just started with a couple.

If merged I can continue.

Reply:
Excellent. Thanks for the good start!